### PR TITLE
New enduser group

### PR DESCRIPTION
**Pull request by tthyer**
- add new user group
- update group names

### CODE REVIEW

File: config/prod/sc-enduser-dev-iam.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+template_path: "remote/sc-enduser-iam.yaml"
+stack_name: "sc-enduser-development-iam"
```
Comment (zaro0508) [inline config/prod/sc-enduser-dev-iam.yaml]: maybe make this match filename?

File: templates/accounts.yaml [hunk 2]
```diff
@@ -19,7 +19,9 @@ Resources:
         - !Ref AWSIAMDeveloperUsersGroup
         - !Ref AWSIAMBillingManagersGroup
         - !ImportValue
-          'Fn::Sub': '${AWS::Region}-sc-enduser-iam-EndUserGroupName'
+          'Fn::Sub': '${AWS::Region}-sc-enduser-iam-ServiceCatalogEndusers-GroupName'
```
Comment (zaro0508) [inline templates/accounts.yaml]: i don't think this matches ${AWS::Region}-${AWS::StackName}-${RoleAndGroupName}-GroupName that's defined here: https://github.com/Sage-Bionetworks/scipoolprod-sc-lib-infra/blob/master/iam/sc-enduser-iam.yaml#L60

File: config/prod/sc-enduser-dev-iam.yaml (hunk 1 above)
Comment (tthyer) [reply config/prod/sc-enduser-dev-iam.yaml]: done

File: templates/accounts.yaml (hunk 2 above)
Comment (tthyer) [reply templates/accounts.yaml]: The pattern is `!Sub '${AWS::Region}-${AWS::StackName}-${RoleAndGroupName}-GroupName'`
This should match the default `RoleAndGroupName` which is "ServiceCatalogEndusers"
 So, this one is right `'${AWS::Region}-sc-enduser-iam-ServiceCatalogEndusers-GroupName'`
But the next line is wrong! Should be `'Fn::Sub': '${AWS::Region}-sc-enduser-development-iam-ServiceCatalogDevelopmentEndusers-GroupName'`
Which is really long but still only 80 characters

File: config/prod/accounts.yaml [hunk 3]
```diff
@@ -3,5 +3,6 @@ stack_name: "accounts"
 dependencies:
   - "prod/essentials.yaml"
   - "prod/sc-enduser-iam.yaml"
+  - "prod/sc-enduser-dev-iam.yaml"
```
Comment (zaro0508) [inline config/prod/accounts.yaml]: now these this filename is out of sync?

File: config/prod/sc-accounts.yaml [hunk 4]
```diff
@@ -3,5 +3,6 @@ stack_name: "sc-accounts"
 dependencies:
   - "prod/essentials.yaml"
   - "prod/sc-enduser-iam.yaml"
+  - "prod/sc-enduser-dev-iam.yaml"
```
Comment (zaro0508) [inline config/prod/sc-accounts.yaml]: same here

File: config/prod/accounts.yaml (hunk 3 above)
Comment (tthyer) [reply config/prod/accounts.yaml]: fixed

File: config/prod/sc-accounts.yaml (hunk 4 above)
Comment (tthyer) [reply config/prod/sc-accounts.yaml]: also fixed

**Pull request merged**